File: 프로그래머스/lv2/42626. 더 맵게/더 맵게.java
import java.util.*;
class Solution {
    public int solution(int[] scoville, int K) {
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        int answer = 0;
        
        for (int i=0; i<scoville.length;i++){
            pq.add(scoville[i]);
        }
        
        while (pq.size()!=1){
            int first = pq.peek();
            if (first>=K){
                break;
            }
            
            int a = pq.poll();
            int b = 0;
            int news = 0;
            if (!pq.isEmpty()){
                b = pq.poll();
                news = a+b*2;
                pq.add(news);
                answer+=1;
            }
            
        }
        // pq.peek() < K 이 경우를 고려 못해서 16번만 오류났었음!!!!!!
        if (pq.size() == 1 && pq.peek() < K){
            answer=-1;
        }
        return answer;
    }
}

File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
from heapq import heappop, heappush

def solution(scoville, K):
    answer = 0
    heap = []
    for s in scoville:
        heappush(heap,s)
    while(True):
        if len(heap)==1 and heap[0]<K:
            answer=-1
            break
        elif heap[0]>=K:
            break
        else:
            h1 = heappop(heap)
            h2 = heappop(heap)
            heappush(heap, h1+h2*2)
            
            answer+=1
    return answer